pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java

```diff
@@ -134,4 +135,18 @@ public File getTmpUntarredPath() {
   public File getSchemasTmpDir() {
     return new File(_schemasTmpDirURI);
   }
+
+  /**
+   * Cleans dataDir by getting rid of unnecessary slashes at the end of dataDir. Needed so other URIs are constructed
+   * as expected.
+   * @param dataDir data directory of pinot segments
+   * @return cleaned data directory
+   */
+  private String cleanDataDir(String dataDir) {
```

Comment:
The name seems to suggest data directory's content would be cleaned/removed. Why not just use existing utility functions such as trim()/strip() to do the same, instead of writing a new one?

Reply:
strip is a replace all or a strip leading/trialing spaces, and trim is just leading/trailing as far as i know, but I only want to remove the last slash. Is there another method I don't know about?

Reply:
Maybe StringUtils.removeEnd(...)?
[https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#removeEnd-java.lang.String-java.lang.String-]

Reply:
thank you!